### PR TITLE
Fixes #2357 - BMP images for avatars

### LINKED ISSUE #2357: BMP images don't work for avatars
If you attempt to upload an avatar that is a BMP image, the upload will fail and it'll tell you to choose a valid file and try again. BMP images are suppose to be valid file type for avatars (https://github.com/mybb/mybb/blob/feature/inc/functions_upload.php#L208) (https://github.com/mybb/mybb/blob/feature/inc/languages/english/usercp.lang.php#L341).

The problem seems to lie at the switch located here: https://github.com/mybb/mybb/blob/feature/inc/functions_upload.php#L300
The $img_type is never set for a BMP file.

### PR DESCRIPTION
**Pull request by Cu8eR**
#2357

### CODE REVIEW

File: inc/functions_upload.php [hunk 1]
```diff
@@ -313,6 +313,9 @@ function upload_avatar($avatar=array(), $uid=0)
 		case "image/x-png":
 			$img_type = 3;
 			break;
+		case "image/bmp":
```
Comment (Destroy666x) [inline inc/functions_upload.php]: That's not the only and not even official MIME.

Comment (Cu8eR) [inline inc/functions_upload.php]: According to Google image/bmp is an official mime type

For other MIME types, I will add them - at least image/x-windows-bmp

Comment (Cu8eR) [inline inc/functions_upload.php]: image/bmp, image/x-bmp, image/x-bitmap, image/x-xbitmap, image/x-win-bitmap, image/x-windows-bmp, image/ms-bmp, image/x-ms-bmp anything else?

File: inc/functions_upload.php [hunk 2]
```diff
@@ -313,6 +313,9 @@ function upload_avatar($avatar=array(), $uid=0)
...
+			$img_type = 4;
```
Comment (Destroy666x) [inline inc/functions_upload.php]: Where exactly did 4 come from?

File: inc/functions_upload.php (hunk 1 above)
Comment (Destroy666x) [inline inc/functions_upload.php]: Only the most common ones should be there: image/bmp, image/x-bmp, image/x-bitmap

Comment (Cu8eR) [inline inc/functions_upload.php]: Fixed! Thank you

File: inc/functions_upload.php [hunk 3]
```diff
@@ -313,6 +313,11 @@ function upload_avatar($avatar=array(), $uid=0)
 		case "image/x-png":
 			$img_type = 3;
 			break;
+		case "image/bmp":
+		case "image/x-bmp":
+		case "image/x-bitmap":
+			$img_type = 4;
```
Comment (Destroy666x) [inline inc/functions_upload.php]: Where exactly did 4 come from? I'm quite sure it's not what's returned..

Comment (Cu8eR) [inline inc/functions_upload.php]: Hmm, I will check and let you know

Comment (Stefan-ST) [inline inc/functions_upload.php]: IMG_WBMP = 8

Comment (Destroy666x) [inline inc/functions_upload.php]: @Stefan-ST isn't it `IMG_BMP`? I get that and a different number.

Comment (Stefan-ST) [inline inc/functions_upload.php]: @Destroy666x I haven't tested it (yet) but `IMG_BMP` not mentioned in the documentation: http://php.net/manual/en/image.constants.php

Comment (PaulBender) [inline inc/functions_upload.php]: I checked, the img_type for a BMP is 6.

Comment (Cu8eR) [inline inc/functions_upload.php]: @PaulBender - thank you! Fixed

Comment (Destroy666x) [inline inc/functions_upload.php]: @Stefan-ST , they're listed under `IMAGETYPE` not `IMAGE`. I also had 6 when I tested.

**Comment by Cu8eR** [comment]
Should be ready to review and merge. Thx all for help

**Pull request merged**